sky/utils/subprocess_utils.py

```diff
@@ -96,29 +96,47 @@ def get_parallel_threads(cloud_str: Optional[str] = None) -> int:
     return max(4, cpu_count - 1) * _get_thread_multiplier(cloud_str)
 
 
+# TODO(andyl): Why this function returns a list of results? Why not yielding?
 def run_in_parallel(func: Callable,
                     args: Iterable[Any],
-                    num_threads: Optional[int] = None) -> List[Any]:
+                    num_threads: Optional[int] = None,
+                    continue_on_error: bool = False) -> List[Any]:
```

Comment:
adding this seems change the return value type from `Value` to `Union[Value, Exception]`, which looks a little bit strange to me. Why do we need this anyway? aren't we already add the try-except in the func?

Reply:
You are right. I thought about not adding a function to catch and log these exceptions, but instead handle all exceptions in `run_in_parallel`. Anyway, it made sense to have this function have an argument like `continue_on_error`.
But then I realized it violated the abstraction concept and would make it harder to print custom logs. So I ended up wrapping the function and logging it.